scala/README.md

```diff
@@ -581,6 +726,61 @@ Additional Remarks
 
 0. Take advantage of simple language features that afford great power but avoid the esoteric ones, especially in the type system.
 
+0. Java `enum`s are very powerful and flexible. Enumeration types in Java can include methods and fields, and enum constants are able to specialize their behavior. For more information, refer to the awesome "Effective Java, 2nd Edition" by Joshua Bloch, in particular Chapter 6, "Enums and Annotations", Item 30: "Use enums instead of int constants".
+
+    Unfortunately, Scala enums are not at feature parity with their Java counterparts. They do not, by default, offer the same flexibility and power. They do not interoperate with Java. They have the same type after erasure (see below). Pattern matching is not exhaustively checked. Play JSON and Scala Pickling do not support enumerations. [Do not expect](https://groups.google.com/forum/#!msg/scala-internals/8RWkccSRBxQ/U4y0XpRJfdQJ) the [situation to change](https://www.reddit.com/r/scala/comments/3aqlhu/is_there_a_voting_mechanism_for_new_scala_features/csgy8i0).
+
+    ```scala
+    object A extends Enumeration { val Abc = Value }
+    object X extends Enumeration { val Xyz = Value }
+
+    object DoesNotCompile {
+      def f(v: A.Value) = "I'm A"
+      def f(v: X.Value) = "I'm X"
+    }
+    ```
+
+    The following idioms bring a little more power to Scala enums:
+
+    ```scala
+    // Custom properties
+    object Suit extends Enumeration {
+      import scala.language.implicitConversions
+
+      type Suit = SuitVal
+
+      implicit def toVal(v: Value) = v.asInstanceOf[SuitVal]
+
+      case class SuitVal private[Suit] (symbol: Char) extends Val
```

Comment:
@matthandlersux The idiom is correct. I didn't invent it, but I made some improvements to it. I've learned it many years ago, on my very first months with Scala, and I've been using it successfully since then.

`Val` is a concrete class that extends `Value`. `Value` is abstract, if we extend it directly, we'd have to provide concrete implementations for its abstract members.